lib/utils/type-tracker/index.ts

```diff
@@ -575,7 +575,7 @@ export function createTypeTracker(context: Rule.RuleContext): TypeTracker {
         if (isObject(tsType)) {
             return UNKNOWN_OBJECT
         }
-        return checker.typeToString(tsType) as TypeInfo
+        return checker?.typeToString(tsType) as TypeInfo
```

Comment:
[`?.` always returns `undefined` if the lhs is nullish](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) (not obvious IMO, but that's how it is). `undefined` isn't compatible with a return type of `TypeInfo | null`.

```suggestion
        return checker ? (checker.typeToString(tsType) as TypeInfo) : null
```